src/layers/core/polygon-layer/polygon-layer.js

```diff
@@ -73,14 +73,14 @@ export default class PolygonLayer extends CompositeLayer {
     }
   }
 
-  _onHoverSubLayer(info) {
-    info.object = (info.object && info.object.feature) || info.object;
-    this.props.onHover(info);
-  }
+  _onPickSubLayer(handler, info) {
+    Object.assign(info, {
+      layer: this,
+      // override object with picked feature
+      object: (info.object && info.object.feature) || info.object
```

Comment:
1. Good catch, this seems to be another bug - `info.object.feature` should be `info.object.object` (reference back to user data from exploded paths)